[WKCI] filter-test-logs can cause that buildbot ends killing an in-progress step because of no output for more than X seconds

clopez opened the pull request:
#### 4f8fc62a1b483c3d3e48caaac0face5733e6f436
<pre>
[WKCI] filter-test-logs can cause that buildbot ends killing an in-progress step because of no output for more than X seconds
<a href="https://bugs.webkit.org/show_bug.cgi?id=302827">https://bugs.webkit.org/show_bug.cgi?id=302827</a>

Reviewed by Nikolas Zimmermann.

The python script filter-test-logs used to send the logs directly to S3
instead of letting buildbot process them can cause that a step times out
because of &quot;command timed out: XXX seconds without output running YYY&quot;
when that step was actually working and printing things to stdout.

This patch modifies filter-test-logs to print a status update to stdout
each 5 minutes with the number of lines processed/filtered, in order to
avoid the buildbot no-output timeout.

* Tools/Scripts/filter-test-logs:
(KeepAliveStdoutProgress):
(KeepAliveStdoutProgress.__init__):
(KeepAliveStdoutProgress.maybe_update_progress):
(filter_jsc_tests):
(filter_layout_tests):
(filter_scan_build):
(filter_webdriver_tests):
(filter_test262):
(filter_minibrowser_tests):

Canonical link: <a href="https://commits.webkit.org/303319@main">https://commits.webkit.org/303319@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e272a52cc6c62adddb5d80265aaaed21c4d74233

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131964 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4456 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42977 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139477 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83861 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133834 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4398 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4218 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/100869 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/68260 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134910 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/3130 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118181 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81660 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/3023 "Passed tests") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/884 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82696 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111774 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36305 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142122 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4126 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36885 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109241 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4207 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3596 "Found 3 new test failures: http/tests/webgpu/webgpu/api/operation/sampling/sampler_texture.html imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video/processing-model/selectors/cue/white-space_nowrap_wrapped.html imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video/processing-model/selectors/cue/white-space_pre.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109411 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3121 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114461 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/57347 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20526 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4179 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32858 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4011 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4271 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4139 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann reviewed (approved):
LGTM, fixing a real problem. Thanks @clopez

clopez committed:
[WKCI] filter-test-logs can cause that buildbot ends killing an in-progress step because of no output for more than X seconds
https://bugs.webkit.org/show_bug.cgi?id=302827

Reviewed by Nikolas Zimmermann.

The python script filter-test-logs used to send the logs directly to S3
instead of letting buildbot process them can cause that a step times out
because of "command timed out: XXX seconds without output running YYY"
when that step was actually working and printing things to stdout.

This patch modifies filter-test-logs to print a status update to stdout
each 5 minutes with the number of lines processed/filtered, in order to
avoid the buildbot no-output timeout.

* Tools/Scripts/filter-test-logs:
(KeepAliveStdoutProgress):
(KeepAliveStdoutProgress.__init__):
(KeepAliveStdoutProgress.maybe_update_progress):
(filter_jsc_tests):
(filter_layout_tests):
(filter_scan_build):
(filter_webdriver_tests):
(filter_test262):
(filter_minibrowser_tests):

Canonical link: https://commits.webkit.org/303319@main
